### ISSUE TITLE
maven_rules breaks --all_incompatible_changes flag

### ISSUE THREAD

**Issue by iphydf** [bazelbuild/bazel#4718]
```
ERROR: /usr/local/home/iphy/.cache/bazel/_bazel_iphy/87c462f2b35b8d3dfe6f6c21308fd127/external/bazel_tools/tools/build_defs/repo/maven_rules.bzl:296:11: Traceback (most recent call last):
	File "/usr/local/home/iphy/.cache/bazel/_bazel_iphy/87c462f2b35b8d3dfe6f6c21308fd127/external/bazel_tools/tools/build_defs/repo/maven_rules.bzl", line 294
		repository_rule(implementation = _maven_jar_impl, <2 more arguments>)
	File "/usr/local/home/iphy/.cache/bazel/_bazel_iphy/87c462f2b35b8d3dfe6f6c21308fd127/external/bazel_tools/tools/build_defs/repo/maven_rules.bzl", line 296, in repository_rule
		_common_maven_rule_attrs + {"repository": attr.string(default = ""), "server": attr.label(default = None)}
The `+` operator for dicts is deprecated and no longer supported. Please use the `update` method instead. You can temporarily enable the `+` operator by passing the flag --incompatible_disallow_dict_plus=false
```

See https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/repo/maven_rules.bzl#L296.

**Comment by cgrushko** [comment]
Jin, can you take a look? (I believe you're the original author)

**Comment by jin** [comment]
@iphydf, what is your Bazel version/commit? This has been fixed by @vladmos in https://github.com/bazelbuild/bazel/commit/ceaed5145f5ad27facd103a79d047cf0e301b4b5#diff-d0efbe851362a2add4518a85575a0c64L296

**Comment by iphydf** [comment]
```
$ bazel version
Build label: 0.10.0-homebrew
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Jan 9 21:07:27 +50057 (1517479996047)
Build timestamp: 1517479996047
Build timestamp as int: 1517479996047
```

**Comment by iphydf** [comment]
Actually, I get the above error on 0.9.0, on 0.10.0 I get a different one:
```
Loading:
Loading: 0 packages loaded
ERROR: /private/var/tmp/_bazel_iphy/178403b66d944445b84ccba31dd03db8/external/io_bazel_rules_scala/third_party/plugin/src/main/BUILD:5:1: in scala_library_for_plugin_bootstrapping rule @io_bazel_rules_scala//third_party/plugin/src/main:dependency_analyzer:
Traceback (most recent call last):
	File "/private/var/tmp/_bazel_iphy/178403b66d944445b84ccba31dd03db8/external/io_bazel_rules_scala/third_party/plugin/src/main/BUILD", line 5
		scala_library_for_plugin_bootstrapping(name = 'dependency_analyzer')
	File "/private/var/tmp/_bazel_iphy/178403b66d944445b84ccba31dd03db8/external/io_bazel_rules_scala/scala/scala.bzl", line 733, in _scala_library_impl
		_lib(ctx, True)
	File "/private/var/tmp/_bazel_iphy/178403b66d944445b84ccba31dd03db8/external/io_bazel_rules_scala/scala/scala.bzl", line 660, in _lib
		_collect_jars_from_common_ctx(ctx)
	File "/private/var/tmp/_bazel_iphy/178403b66d944445b84ccba31dd03db8/external/io_bazel_rules_scala/scala/scala.bzl", line 621, in _collect_jars_from_common_ctx
		collect_jars(((ctx.attr.deps + auto_deps) + e...), ...)
	File "/private/var/tmp/_bazel_iphy/178403b66d944445b84ccba31dd03db8/external/io_bazel_rules_scala/scala/scala.bzl", line 597, in collect_jars
		_collect_jars_when_dependency_analyzer_is_off(dep_targets)
	File "/private/var/tmp/_bazel_iphy/178403b66d944445b84ccba31dd03db8/external/io_bazel_rules_scala/scala/scala.bzl", line 549, in _collect_jars_when_dependency_analyzer_is_off
		compile_jars += java_provider.compile_jars
`+` operator on a depset is forbidden. See https://docs.bazel.build/versions/master/skylark/depsets.html for recommendations. Use --incompatible_depset_union=false to temporarily disable this check.
ERROR: Analysis of target '//jvm-toxcore-c:AudioReceiveFrameCallbackTest' failed; build aborted: Analysis of target '@io_bazel_rules_scala//third_party/plugin/src/main:dependency_analyzer' failed; build aborted
INFO: Elapsed time: 0.196s
FAILED: Build did NOT complete successfully (0 packages loaded)
```

**Comment by iphydf** [comment]
Ah, and that's scala related, not maven. Ok, confirmed that it's fixed in 0.10.0.
